R/pkg/R/sparkR.R

```diff
@@ -283,6 +283,10 @@ sparkR.session <- function(
   enableHiveSupport = TRUE,
   ...) {
 
+  if (utils::compareVersion(paste0(R.version$major, ".", R.version$minor), "3.4.0") == -1) {
+    warning("R prior to version 3.4 is deprecated as of Spark 3.0.")
+  }
```

Comment:
@felixcheung, I updated the condition to check major version as well.

BTW, if we add it in .First in general.R, looks it's not going to print out the warnings in the SparkR shell. When we drop PySpark, we did in a similar place (https://github.com/apache/spark/pull/15733/files#diff-01e513a19a2a6f31ecaabd3dd6ac12d9R192).